A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/EICAS_Common.js

```diff
@@ -77,23 +77,23 @@ class EICASCommonDisplay extends Airliners.EICASTemplateElement {
         }
     }
     refreshGrossWeight(_force = false) {
-        let gw = 0;
         let isInMetric = BaseAirliners.unitIsMetric(Aircraft.A320_NEO);
-        if (isInMetric) {
-            gw = Math.round(SimVar.GetSimVarValue("TOTAL WEIGHT", "kg"));
-            if (this.gwUnit)
-                this.gwUnit.textContent = "KG";
-        }
-        else {
-            gw = Math.round(SimVar.GetSimVarValue("TOTAL WEIGHT", "lbs"));
-            if (this.gwUnit)
-                this.gwUnit.textContent = "LBS";
+        let unit = isInMetric ? "kg" : "lbs";
+        let gw = 0;
+        let payloadWeight = 0;
+        let payloadCount = SimVar.GetSimVarValue("PAYLOAD STATION COUNT", "number");
+        let fuelWeight = Math.round(SimVar.GetSimVarValue("FUEL TOTAL QUANTITY WEIGHT", unit));
+        let emptyWeight = Math.round(SimVar.GetSimVarValue("EMPTY WEIGHT", unit));
+        for (var i = 1; i <= payloadCount; i++) {
+            payloadWeight += Math.round(SimVar.GetSimVarValue(`PAYLOAD STATION WEIGHT:${i}`, unit));
         }
+        gw = emptyWeight + fuelWeight + payloadWeight;
```

Comment:
I mentioned this before but the weight should only be rounded down once so we retain the decimals. If we don't then there's a chance for display errors.

Reply:
> I mentioned this before but the weight should only be rounded down once so we retain the decimals. If we don't then there's a chance for display errors.

ok. Let me round the gw before set the value to the UI. This is my first PR for the flybywiresim project. Sorry for that.

Reply:
It's no biggie!

Reply:
I pushed new code base on your review @UncleClapton